From a5a38f68b9e26b4bdf41455fe574e7b3c185ac71 Mon Sep 17 00:00:00 2001
From: Fang Lijun <fanglijun3@huawei.com>
Date: Fri, 3 Dec 2021 18:15:58 +0800
Subject: [PATCH] arm64/numa: cdm: Cacheline aligned cdmmask to improve
 performance
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: a5a38f68b9e26b4bdf41455fe574e7b3c185ac71


ascend inclusion
category: Bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I4JMLR
CVE: NA

--------------

Enable COHERENT_DEVICE will degrade performance, Hackbench test
time (Pipe_Process_Number=800) from 0.3 to 1.8. When the cdmmask
was cacheline aligned, it will be improved as same as disabled
COHERENT_DEVICE.

Signed-off-by: Fang Lijun <fanglijun3@huawei.com>
Reviewed-by: Weilong Chen <chenweilong@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/mm/numa.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/mm/numa.c b/arch/arm64/mm/numa.c
index b2260bb53691..9a2e29a3a597 100644
--- a/arch/arm64/mm/numa.c
+++ b/arch/arm64/mm/numa.c
@@ -26,7 +26,7 @@ static u8 *numa_distance;
 bool numa_off;
 
 #ifdef CONFIG_COHERENT_DEVICE
-nodemask_t cdmmask;
+nodemask_t __cacheline_aligned cdmmask;
 
 inline int arch_check_node_cdm(int nid)
 {
-- 
2.26.2

